src/generators.py

```diff
@@ -7,6 +7,19 @@ def range_generator(start, end=None):
         start += 1
 
 
+def iterator(generator):
+    generated_list = []
+    try:
+        while True:
```

Comment:
It's better to have as little code in `try` as possible, so you should move `while` out of `try`